3 - Assignment/240409 - Rust Basic + Roguelike Game, Week 2/generic-logger/src/lib.rs

```diff
@@ -18,7 +18,18 @@ fn do_things(logger: &impl Logger) {
     logger.log(2, "Uhoh");
 }
 
-// TODO: Define and implement `VerbosityFilter`.
+struct VerbosityFilter{
+    max_verbosity:u8,
+    inner:StderrLogger
+}
+
+impl Logger for VerbosityFilter{
+    fn log(&self, verbosity: u8, message: impl Display) {
+        if verbosity <= self.max_verbosity{
+            self.inner.log(verbosity,message)
+        }
+    }
+}
```

Comment:
`cargo fmt`를 통해 코드 스타일을 맞춰주세요.

```suggestion
struct VerbosityFilter {
    max_verbosity: u8,
    inner: StderrLogger
}

impl Logger for VerbosityFilter {
    fn log(&self, verbosity: u8, message: impl Display) {
        if verbosity <= self.max_verbosity {
            self.inner.log(verbosity,message);
        }
    }
}
```